File: src/test/resources/oracle/commits/rm-action-test-c6e2f18c7ba5526bd698189c6b78f094fcd7b971/OrderProcessor.java
public class OrderProcessor {
    private final ReceiptFormatter formatter = new ReceiptFormatter();

    public String buildReceipt(String buyerName, int unitPrice, int quantity, int discount) {
        int total = calculateTotal(unitPrice, quantity, discount);
        int subtotal = unitPrice * quantity;
        int discountedSubtotal = subtotal - discount;
        int tax = (int) (discountedSubtotal * new PricingPolicy().taxRate);

        String header = formatter.createReceiptHeader(buyerName);
        double dollars = total / 100.0;
        String status = describePaymentStatus(total, discount);

        return header + "\n"
                + "Subtotal: " + subtotal + "\n"
                + "Discount: " + discount + "\n"
                + "Tax: " + tax + "\n"
                + "Total: $" + dollars + "\n"
                + status;
    }

    private int calculateTotal(int unitPrice, int quantity, int discount) {
        int subtotal = unitPrice * quantity;
        int discountedSubtotal = subtotal - discount;
        int tax = (int) (discountedSubtotal * new PricingPolicy().taxRate);
        return discountedSubtotal + tax;
    }

    private String describePaymentStatus(int total, int discount) {
        if (total > 0 && discount > 0) {
            return "Discount applied";
        }
        if (total > 0) {
            return "Payment required";
        }
        return "No payment required";
    }
}

File: src/test/resources/oracle/commits/rm-action-test-60d0d6f35eeed488560c247c3b4fa29dd3ac39d4/kotlin/OrderProcessor.kt
class OrderProcessor {
    private val formatter = ReceiptFormatter()

    fun buildReceipt(buyerName: String, unitPrice: Int, quantity: Int, discount: Int): String {
        val total = calculateTotal(unitPrice, quantity, discount)
        val subtotal = unitPrice * quantity
        val discountedSubtotal = subtotal - discount
        val tax = (discountedSubtotal * PricingPolicy().taxRate).toInt()

        val header = formatter.createReceiptHeader(buyerName)
        val dollars = total / 100.0
        val status = describePaymentStatus(total, discount)

        return header + "\n" +
                "Subtotal: " + subtotal + "\n" +
                "Discount: " + discount + "\n" +
                "Tax: " + tax + "\n" +
                "Total: $" + dollars + "\n" +
                status
    }

    private fun calculateTotal(unitPrice: Int, quantity: Int, discount: Int): Int {
        val subtotal = unitPrice * quantity
        val discountedSubtotal = subtotal - discount
        val tax = (discountedSubtotal * PricingPolicy().taxRate).toInt()
        return discountedSubtotal + tax
    }

    private fun describePaymentStatus(total: Int, discount: Int): String {
        if (total > 0 && discount > 0) {
            return "Discount applied"
        }
        if (total > 0) {
            return "Payment required"
        }
        return "No payment required"
    }
}
